src/marlin_stubs/G26.cpp

```diff
@@ -330,9 +336,27 @@ void FirstLayer::print_shape_2() {
 }
 
 void PrusaGcodeSuite::G26() {
-    if (all_axes_known()) { /// checks if axes are calibrated (homed) before
-        FirstLayer fl;
-        //fl.print_shape_1();
-        fl.print_shape_2();
+    // is filament selected
+    if (Filaments::Current().response == Response::Cooldown) {
+        return;
     }
+
+    FirstLayer fl;
+
+    const int temp_nozzle = Filaments::Current().nozzle;
+
+    // nozzle temperature print
+    thermalManager.setTargetHotend(temp_nozzle, 0);
+    marlin_server_set_temp_to_display(temp_nozzle);
+    thermalManager.wait_for_hotend(0, false);
+
+    //fl.print_shape_1();
+    fl.print_shape_2();
+
+    // nozzle temperature 0
+    thermalManager.setTargetHotend(0, 0);
+    marlin_server_set_temp_to_display(0);
+
+    // bed temperature 0
```

Comment:
Redundant comments. Setting target hotend temp and bed temperature to 0 is obvious and doesn't need to be commented.